### ISSUE TITLE
disapearing first word of the line after newline+space highlighting and enter

### ISSUE THREAD

**Issue by fabianegli** [DSpeckhals/python-indent#42]
The issue comes up when highlight from the end of the line above to the last leading white space on the line and hit enter. 

Lets start with the following code block with too much indentation:
```
def a():
    def function_with_lots_of_params(param_1, param_2,
                                         param_3, param_4,
                                         very_long_parameter_name,
                                         param_6)
```
Highlight from right after the comma of param_2 and highlight all white space up to but not including param_3 and hit enter. And you get this:

```
def a():
    def function_with_lots_of_params(param_1, param_2,
		    			3, param_4,
                                         very_long_parameter_name,
                                         param_6)
```
The 'param_3' is clearly missing the 'param_' string and this is bad. 

It works if the 'function_with_lots_of_params' is not indented itself, but even more text gets eaten  #when the indent of 'function_with_lots_of_params' increases.

Stuff also disappears when the cursor is is after 'param_1,' and I hit enter.

**Comment by fabianegli** [comment]
I am not sure anymore that this issue is caused by this package. I first saw it appear after I installed python-indent, but it did not vanish when I uninstalled it. I am very puzzled about this behavior of my atom since I would never expect it to delete anything at all without my expressed intent.

**Comment by kbrose** [comment]
@fabianegli Your example works fine for me on Windows and Ubuntu. Looks like it's probably another package you've got installed?

I would also look at your python language settings - your second block of code you've pasted has tabs in it instead of spaces...

**Comment by kbrose** [comment]
Hmm looks like that's it - if you're using tabs then it can get messed up.

**Comment by kbrose** [comment]
@DSpeckhals it looks like we should probably add a check for what tab type is being used and not do anything if tabs are used (but why would you want PEP8 style indentation with tabs, that's just not possible...). Although I also thought we overwrote that with spaces regardless of what the user chose... maybe this broke in recent Atom update?

**Comment by kbrose** [comment]
@fabianegli short term fix - set your editor to use "soft tabs", i.e. spaces, instead of tabs.

**Comment by DSpeckhals** [comment]
This issue was fixed by #43

**Comment by fabianegli** [comment]
Thank you all very much for the fast fix and pointing out the spaces/tabs issue! Also thanks @kbrose  for the hint to the "soft tabs".
